src/map.c

```diff
@@ -1,754 +1,592 @@
-/*
```

Comment:
Keep this copyright notice. And mention the source was heavily modified from rb.h of jemalloc.